### PR TITLE
SplitButton: updated border right token for primary variant

### LINKED ISSUE #28532: [Bug]: Split button primary variant, border right has wrong token color assignment
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
Fluent v9
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://master--628d031b55e942004ac95df1.chromatic.com/?path=/docs/components-button-splitbutton--default

## Actual Behavior
image
image

## Expected Behavior
Please assign` colorNeutralStrokeOnBrand`
image

so in dark theme the stroke is dark
image

cc: @khmakoto @jurokapsiar @miro

### Logs

_No response_

### Requested priority

High

### Products/sites affected

Teams

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by kkakroo**
## Previous Behavior
Incorrect token was assigned to right border primary variant for split button

## New Behavior
Correct token assigned to right border primary variant for split button

## Related Issue(s)

- Fixes #28532

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28555/perf-test-react-components/Avatar-mount_base.html">592</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28555/perf-test-react-components/Avatar-mount.html">623</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28555/perf-test-react-components/Button-mount_base.html">294</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28555/perf-test-react-components/Button-mount.html">313</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28555/perf-test-react-components/Field-mount_base.html">1073</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28555/perf-test-react-components/Field-mount.html">1097</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28555/perf-test-react-components/FluentProvider-mount_base.html">655</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28555/perf-test-react-components/FluentProvider-mount.html">653</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28555/perf-test-react-components/FluentProviderWithTheme-mount_base.html">78</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28555/perf-test-react-components/FluentProviderWithTheme-mount.html">81</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28555/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">78</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28555/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">62</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28555/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">75</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28555/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">73</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28555/perf-test-react-components/InfoButton-mount_base.html">17</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28555/perf-test-react-components/InfoButton-mount.html">9</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28555/perf-test-react-components/MakeStyles-mount_base.html">890</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28555/perf-test-react-components/MakeStyles-mount.html">905</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28555/perf-test-react-components/Persona-mount_base.html">1684</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28555/perf-test-react-components/Persona-mount.html">1589</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28555/perf-test-react-components/SpinButton-mount_base.html">1278</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28555/perf-test-react-components/SpinButton-mount.html">1313</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                    |     Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                  Change |
| :--------------------------------------------------------------------------------------------------- | ---------------------------: | ---------------------------: | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr> | `49.211 kB`<br />`12.232 kB` | `49.183 kB`<br />`12.231 kB` | `-28 B` <br />`-1 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |         Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-alert</samp> <br /> <abbr title='bundle-size/Alert.fixture.js'>Alert</abbr>                                                                                       | `82.565 kB`<br />`21.726 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr>                                                                                    |  `36.792 kB`<br />`9.472 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr>                                                                    | `43.946 kB`<br />`10.952 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr>                                                                            | `40.978 kB`<br />`10.669 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr>                                                                        | `55.074 kB`<br />`11.368 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.All.fixture.js'>Card - All</abbr>                                                                                |  `89.36 kB`<br />`25.284 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.fixture.js'>Card</abbr>                                                                                          | `83.778 kB`<br />`23.685 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardFooter.fixture.js'>CardFooter</abbr>                                                                              |    `9.24 kB`<br />`3.907 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardHeader.fixture.js'>CardHeader</abbr>                                                                              |  `11.652 kB`<br />`4.741 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardPreview.fixture.js'>CardPreview</abbr>                                                                            |  `10.048 kB`<br />`4.254 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              | `65.175 kB`<br />`17.915 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `210.68 kB`<br />`58.781 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            | `36.409 kB`<br />`12.003 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |   `6.473 kB`<br />`2.196 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/df56ccd0c070c01191acc1af2ddf61f458228c0b'>df56ccd0c070c01191acc1af2ddf61f458228c0b</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: change/@fluentui-react-button-900164ea-88e7-4c4d-8bc8-36ec6d3e2619.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "fix: updated right border token in split button",
```
Comment (khmakoto) [inline change/@fluentui-react-button-900164ea-88e7-4c4d-8bc8-36ec6d3e2619.json:3]:
```suggestion
  "comment": "fix: updated right border token in primary variant of split button",
```

**Review by khmakoto — APPROVED** [review]

File: change/@fluentui-react-button-900164ea-88e7-4c4d-8bc8-36ec6d3e2619.json (hunk 1 above)
Comment (kkakroo) [reply change/@fluentui-react-button-900164ea-88e7-4c4d-8bc8-36ec6d3e2619.json]: Hi @khmakoto, updated as requested.

**Review by khmakoto — APPROVED** [review]

**Pull request merged**